x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_key_flyout.tsx

```diff
@@ -252,41 +333,45 @@ export const CreateApiKeyFlyout: FunctionComponent<CreateApiKeyFlyoutProps> = ({
           <EuiFormFieldset>
             <EuiSwitch
               label={i18n.translate(
-                'xpack.security.accountManagement.createApiKey.customExpirationLabel',
+                'xpack.security.accountManagement.apiKeyFlyout.customExpirationLabel',
                 {
                   defaultMessage: 'Expire after time',
                 }
               )}
               checked={!!form.values.customExpiration}
               onChange={(e) => form.setValue('customExpiration', e.target.checked)}
+              disabled={readonly || !!selectedApiKey}
               data-test-subj="apiKeyCustomExpirationSwitch"
             />
             {form.values.customExpiration && (
               <>
                 <EuiSpacer size="m" />
                 <EuiFormRow
                   error={form.errors.expiration}
-                  isInvalid={form.touched.expiration && !!form.errors.expiration}
+                  isInvalid={form.touched.expiration && !!form.errors.expiration && !selectedApiKey}
                   label={i18n.translate(
-                    'xpack.security.accountManagement.createApiKey.customExpirationInputLabel',
+                    'xpack.security.accountManagement.apiKeyFlyout.customExpirationInputLabel',
                     {
                       defaultMessage: 'Lifetime (days)',
```

Comment:
Can we remove (days) here as the word "days" is included in the field?

Reply:
@gchaps Just the days in the label? or on the input box as well?

Reply:
Just the days in the label.

Reply:
`(days)` removed in latest commit!